src/software/thunderscope/common/proto_plotter.py

```diff
@@ -112,11 +113,12 @@ def refresh(self):
                     if name not in self.plots:
                         self.data_x[name] = deque([], self.buffer_size)
                         self.data_y[name] = deque([], self.buffer_size)
+
+                        # Ensure hue has sufficient contrast
+                        self.color_hue = (self.color_hue + randint(50, 310)) % 360
                         self.plots[name] = self.win.plot(
-                            pen=QtGui.QColor(
-                                random.randint(100, 255),
-                                random.randint(100, 255),
-                                random.randint(100, 255),
+                            pen=QtGui.QColor.fromHsl(
+                                self.color_hue, randint(75, 100), randint(60, 90)
                             ),
```

Comment:
nit: The colors right now are a bit hard to see. 
![image](https://github.com/UBC-Thunderbots/Software/assets/28585597/a93d647c-da00-4e8c-9544-8a2ab0aae076)

I think we should maximize the saturation and have the luminance at 50% to maximize the color's contract
```suggestion
                            pen=QtGui.QColor.fromHsl(
                                self.color_hue, 255, 255 // 2
                            ),
```
![image](https://github.com/UBC-Thunderbots/Software/assets/28585597/405c76bc-d703-4f07-9e5c-35bdb3424e41)

I got the idea from playing around with this: https://hslpicker.com/#e433ff

Reply:
I didn't realize that `fromHsl` takes values in `[0,255]` for saturation and lightness, instead of `[0,100]`. This certainly makes the colors easier to see.

Reply:
I experimented with the values some more, and I think that higher lightness (for example, 80%) results in better contrast with the background. With 50% lightness, red and blue end up quite dark.